app/assets/javascripts/checkbox_control.es6

```diff
@@ -0,0 +1,50 @@
+$(function() {
+
+  /*
+    Usage:
+      Required: A group of checkboxes,
+        and another control checkbox outside the group
+
+      Enclose the checkbox group with a div of class "deselect-group"
+        and set it's "data-deselect-ctrl" to the jquery identifier for the control checkbox
+  */
+  $('.deselect-group').each(function() {
+    let container = $(this);
+    let control = $(container.data('deselect-ctrl'));
+    let checkboxMemory = [];
+    let checkboxes = container.find("input:checkbox");
+    let controlExists = control.length
+
+    if(controlExists) {
```

Comment:
would a return early statement be better?
```javascript
if(!control.length) return
```
I think the `controlExists` variable is unnecessary and using `control.length` is clear enough but feel free to keep it

Reply:
My lack of familiarity with JavaScript idioms. `thing.length` returning `nil` if `thing` doesn't exist on page always makes me have to think "what's that doing?".

I hadn't realised you could do `if(something)) return`. I'll try that now.